File: scripts/debug/data/backup_20250722_162311/test_sensor_raw_data.py
#!/usr/bin/env python3
"""Test what raw data sensors are returning"""

import requests
import base64
from datetime import datetime, timedelta

def test_sensor(name, ip, port, username, password):
    """Test a single sensor"""
    print(f"\n📡 Testing {name} ({ip}:{port})...")
    
    now = datetime.now()
    one_hour_ago = now - timedelta(hours=1)
    
    # Format dates
    time_start = one_hour_ago.strftime("%Y-%m-%d-%H:%M:%S")
    time_end = now.strftime("%Y-%m-%d-%H:%M:%S")
    
    url = f"http://{ip}:{port}/dataloader.cgi"
    params = {
        'dw': 'vcalogcsv',
        'report_type': '0',
        'statistics_type': '3',
        'linetype': '31',
        'time_start': time_start,
        'time_end': time_end
    }
    
    auth_string = base64.b64encode(f"{username}:{password}".encode()).decode('utf-8')
    headers = {
        'Authorization': f'Basic {auth_string}'
    }
    
    try:
        response = requests.get(url, params=params, headers=headers, timeout=30)
        if response.ok:
            lines = response.text.strip().split('\n')
            print(f"   ✅ Got {len(lines)-1} data rows")
            
            # Show raw CSV header and first few rows
            if len(lines) > 0:
                print(f"\n   CSV Header: {lines[0]}")
                print("\n   Sample data rows (raw):")
                for i, line in enumerate(lines[1:6], 1):  # Show first 5 data rows
                    print(f"   Row {i}: {line}")
                    
                # Parse and show interpreted data
                print("\n   Interpreted data:")
                for line in lines[1:4]:  # Show first 3 rows interpreted
                    parts = line.split(',')
                    if len(parts) >= 17:
                        timestamp = parts[0]
                        # The columns are: timestamp, end_time, device_mac, device_name, sub_area,
                        # line1_in, line1_out, line1_name, line2_in, line2_out, line2_name,
                        # line3_in, line3_out, line3_name, line4_in, line4_out, line4_name
                        print(f"\n   {timestamp}:")
                        print(f"     Line 1: IN={parts[5]}, OUT={parts[6]} ({parts[7]})")
                        print(f"     Line 2: IN={parts[8]}, OUT={parts[9]} ({parts[10]})")
                        print(f"     Line 3: IN={parts[11]}, OUT={parts[12]} ({parts[13]})")
                        print(f"     Line 4: IN={parts[14]}, OUT={parts[15]} ({parts[16]})")
        else:
            print(f"   ❌ Error: {response.status_code}")
            print(f"   Response: {response.text[:200]}...")
    except Exception as e:
        print(f"   ❌ Error: {e}")

def main():
    print("🔍 Testing raw sensor data...")
    print("=" * 50)
    
    # Test each sensor
    sensors = [
        ("J&J Arrábida", "176.79.62.167", 2102, "admin", "grnl.2024"),
        ("Omnia Guimarães", "93.108.96.96", 21001, "admin", "grnl.2024"),
        ("Omnia Almada", "188.37.175.41", 2201, "admin", "grnl.2024"),
        ("Omnia NorteShopping", "188.37.124.33", 21002, "admin", "grnl.2024")
    ]
    
    for sensor in sensors:
        test_sensor(*sensor)

if __name__ == "__main__":
    main()

File: scripts/debug/test-sensor-raw-data.js
#!/usr/bin/env node

/**
 * Test script to see raw sensor data format
 */

const http = require('http');

async function testSensorData() {
  const sensor = {
    sensor_ip: '77.54.194.154',
    sensor_port: 8888,
    sensor_name: 'J&J-ARR-01-PC'
  };

  // Get current time
  const now = new Date();
  const oneHourAgo = new Date(now.getTime() - 60 * 60 * 1000);
  
  const formatDate = (date) => {
    const pad = (n) => n.toString().padStart(2, '0');
    return `${date.getFullYear()}-${pad(date.getMonth() + 1)}-${pad(date.getDate())}-${pad(date.getHours())}:${pad(date.getMinutes())}:${pad(date.getSeconds())}`;
  };

  console.log('Fetching data from sensor...');
  console.log(`Time range: ${formatDate(oneHourAgo)} to ${formatDate(now)}`);

  const path = `/dataloader.cgi?dw=vcalogcsv&report_type=0&statistics_type=3&linetype=31&time_start=${formatDate(oneHourAgo)}&time_end=${formatDate(now)}`;
  
  const options = {
    hostname: sensor.sensor_ip,
    port: sensor.sensor_port,
    path: path,
    method: 'GET',
    headers: {
      'Authorization': 'Basic ' + Buffer.from('admin:grnl.2024').toString('base64')
    },
    timeout: 30000
  };

  return new Promise((resolve, reject) => {
    const req = http.request(options, (res) => {
      let data = '';
      res.on('data', (chunk) => data += chunk);
      res.on('end', () => {
        if (res.statusCode === 200) {
          console.log('\nRaw CSV data:');
          console.log('=============');
          const lines = data.trim().split('\n');
          // Show first 10 lines
          lines.slice(0, 10).forEach((line, i) => {
            console.log(`Line ${i}: ${line}`);
          });
          
          console.log('\nParsing timestamps from data rows:');
          console.log('==================================');
          // Parse a few data rows to see the timestamp format
          for (let i = 1; i < Math.min(6, lines.length); i++) {
            const parts = lines[i].split(',').map(p => p.trim());
            if (parts.length >= 2) {
              console.log(`\nRow ${i}:`);
              console.log(`  Start time: "${parts[0]}"`);
              console.log(`  End time: "${parts[1]}"`);
              
              // Try to parse
              try {
                const start = new Date(parts[0].replace(/\//g, '-'));
                const end = new Date(parts[1].replace(/\//g, '-'));
                console.log(`  Parsed start: ${start.toISOString()}`);
                console.log(`  Parsed end: ${end.toISOString()}`);
                console.log(`  Start minute:second = ${start.getMinutes()}:${start.getSeconds()}`);
                console.log(`  End minute:second = ${end.getMinutes()}:${end.getSeconds()}`);
              } catch (e) {
                console.log(`  Parse error: ${e.message}`);
              }
            }
          }
          
          resolve(data);
        } else {
          reject(new Error(`HTTP ${res.statusCode}: ${data}`));
        }
      });
    });
    
    req.on('error', reject);
    req.on('timeout', () => {
      req.destroy();
      reject(new Error('Request timeout'));
    });
    
    req.end();
  });
}

testSensorData()
  .then(() => console.log('\nDone!'))
  .catch(err => console.error('Error:', err.message));